nixos/modules/config/zram.nix

```diff
@@ -125,6 +176,13 @@ in
         })
         devices);
 
-  };
+    boot.kernel.sysctl = lib.mkIf cfg.recommendedTuningSettings {
+        "vm.swappiness" = 150;
+        "vm.watermark_boost_factor" = 0;
+        "vm.watermark_scale_factor" = 125;
+        "vm.page-cluster" = 0;
```

Comment:
I had a PR for these too https://github.com/NixOS/nixpkgs/pull/268121
You might want to copy the zstd conditional.

Reply:
Oh I didn't know about this PR, sorry!

Hmm, you added that page-cluster conditional because of that [Reddit benchmark of algorithms you linked in there](https://web.archive.org/web/20231027213550/https://old.reddit.com/r/Fedora/comments/mzun99/new_zram_tuning_benchmarks/). The author seems to note a minor decompression throughput gain with lz4 however with nearly double latency, but I would say that latency is more important here for interactivity.

Also the author brings home my point that the kernel defaults are assuming a non-volatile storage backed swap device / file.